generators/delimited/main.py

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env python3
+import sys
+import os
+import csv
+from time import strftime
+from unicode_csv import UnicodeDictReader
+
+# Minimum possible header.
+HEADER = (
+    'entity',
+    'attribute',
+    'value',
+)
+
+def generate(file_name, encoding, delimiter, identifier):
+    # Creating a CSV writer that feeds to stdout.
+    w = csv.writer(sys.stdout)
+
+    # Write the header to start.
+    w.writerow(HEADER)
+
+    # Create each fact
+    with open(file_name, 'rU') as f:
+        sniff = 1024
+        dialect = None
+
+        # Infer various properties about the file
+        # Sample the file to determine the dialect
+        sample = '\n'.join([l for l in f.readlines(sniff)])
+        f.seek(0)
+
+        # Determine dialect
+        sniffer = csv.Sniffer()
+        dialect = sniffer.sniff(sample)
+
+        reader = UnicodeDictReader(f, dialect=dialect, delimiter=delimiter, encoding=encoding)
+        if not identifier:
+            identifier = reader.fieldnames[0]
+
+        for line in reader:
+            # Assemble the identifier's fieldnames into a single string for output.
+            id = ''
+            if identifier is not None:
+                id = build_id(line, identifier)
+            else:
+                id = line[identifier]
+
+            # Use the identifier for each entry as the entity for the new fact(s).
+            # Use the remaining fields as attributes for that entity.
+            for key, value in line.items():
+                w.writerow((
+                    id,
+                    key,
+                    value,
+                ))
+
+    # Flush the remaining bits.
+    sys.stdout.flush()
+
+
+def main():
+    usage = """Origins Delimited Generator.
+
+    Usage:
+        delimited <file_name> [--encoding=<encoding>] [--delimiter=<delimiter>] [--identifier=<indentifier>]
+
+    Options:
+        -h --help  Show usage info.
+        --encoding=<encoding>  The text encoding type of the file being read.
+        --delimiter=<delimiter>  The character used to separate items in the csv file.
+        --identifier=<identifier>  A comma-delimited (no spaces) list of fieldnames which make up a unique identifier for an entry in the file. e.g. --identifier=first,last,birthdate
+    """ #noqa
+
+    from docopt import docopt
+
+    # Trim 'origins generate' from the line.
+    argv = sys.argv[2:]
```

Comment:
I had left a comment before, but it must have gotten lost. Trim only the first argument since `origins generate` will handle trimming it's own commands.